### PR TITLE
Patches Link and adds tests

### PR DESCRIPTION
**Pull request by elliotBraem**
There is a mismatch of the web component's implementation of Link and NearSocial/viewer's
-- it was fixed in [this patch](https://github.com/NearSocial/viewer/commit/528a44d69206b066e3becf7567a25cd32e478139) because of a vulnerability when passing invalid strings

Fixing it here and added tests to validate

### CODE REVIEW

File: playwright-tests/testUtils.js [hunk 1]
```diff
@@ -40,3 +42,28 @@ export const escapeHtml = (html) => {
...
+export const useCode = async (page, filePath, props) => {
+  const fullPath = path.join(__dirname, "code", filePath);
+  try {
+    const code = fs.readFileSync(fullPath, "utf8");
+    const initialProps = props ? JSON.stringify(props) : "";
+
+    // Set code and initialProps attribute
+    await page.evaluate(
+      ({ code, initialProps }) => {
+        const viewer = document.querySelector("near-social-viewer");
```
Comment (petersalomonsen) [inline playwright-tests/testUtils.js:55]: The original code was creating a new `new-social-viewer`element, while this new version is changing the existing. Do we know that changing attributes of the existing element triggers the same behaviour as creating a new?

Comment (elliotBraem) [reply playwright-tests/testUtils.js:55]: I had created the useCode function to support more complex cases, make code injection easier, and to address an issue that was arising while working on the Near.call tests: I noticed that modifying an existing element maintained the selectorPromise and allowed auth storage stages to work ~better~ (there are still some challenges-- and this happened during the "BOS is dead" talks, so had lost some motivation to support the regression tests -- I will write a comment there after this to detail these challenges).

Creating a new element via evaluate removes the selectorPromise -- how can we mock this?

I don't think this matters so much for the link tests or replacing evaluate in the vm/customelements test -- the tests pass the same before and after -- and there is still an existing test creating a new element.

I can write some tests for "code" attribute that verifies it works whether modifying an existing or creating new -- would that remedy the concern?

Comment (petersalomonsen) [reply playwright-tests/testUtils.js:55]: yes that would fix it. probably not an issue at all, but then we are absolutely sure 👍

**Review by petersalomonsen — APPROVED** [review]

File: playwright-tests/testUtils.js (hunk 1 above)
Comment (elliotBraem) [reply playwright-tests/testUtils.js]: Added tests for code and initial props

"should render code with props when creating a new element"
"should render code with props when changing attributes of an existing element"

**Pull request merged**